backend/app/cmd/server_test.go

```diff
@@ -345,7 +348,18 @@ func TestServerApp_Failed(t *testing.T) {
 }
 
 func TestServerApp_Shutdown(t *testing.T) {
```

Comment:
I can't find it, may you give some details?

Reply:
`go test -race -count 1 ./...` shows race conditions in `TestServerApp_Shutdown`, `TestSearch_OtherFields` and `TestSearch_SiteMux`.

Reply:
Data race happens because I increment `shutdownWait` inside `indexDocumentWorker` and this write wasn't synchronized with `Wait` in `Close` method. So, I've move increment [outside](https://github.com/umputun/remark42/pull/757/files#diff-c78bc7caba2f8fa696a1de6869a8a875R92) of function, but caller (same module) should remember to call it, is it ok?


Also I wonder why there is no race with [this](https://github.com/umputun/remark42/pull/757/files#diff-9210fd29d3e63f5bdab3f203d850af54R89) access to shutdownWait. Is it true that it synchronizes via queueNotifier ?

Reply:
The link you provided is not valid, please quote the code. Race condition is gone as far as I can see.